### PR TITLE
scrubber: handle initdb files, fix an issue with prefixes

### PR DESCRIPTION
**Pull request by jcsp**
- The code for calculating the prefix in the bucket was expecting a trailing slash (as it is in the tests), but that's an awkward expectation to impose for use in the field: make the code more flexible by only trimming a trailing character if it is indeed a slash.
- initdb archives were detected by the scrubber as malformed layer files.  Teach it to recognize and ignore them.

### CODE REVIEW

**Comment by arpad-m** [comment]
I wonder if we should just have the scrubber delete them instead of ignoring them. The safekeepers are moving towards deletion of deleted timelines, at least according to [this](https://neondb.slack.com/archives/C059ZC138NR/p1702040824637559?thread_ts=1702007815.780059&cid=C059ZC138NR) msg in slack. We can always un-delete should the user feel that they pressed the wrong button.

So I'm wondering about making the pageserver delete the initdb during deletion also, or maybe implementing a separate "initdb deletion" endpoint or something to prevent needing a parameter to whether preserve the initdb for a later re-creation or not.

We should also watch out for the cases which we want to disaster recover, we should not have the scrubber delete the file while the DR process needs them... not sure how to address that point.

**Comment by jcsp** [comment]
> I wonder if we should just have the scrubber delete them instead of ignoring them. 

Yes, that would be an extra thing for the `find-garbage`/`purge-garbage` commands, perhaps some logic like removing the initdb for deleted timelines after some time period.  The `scan-metadata` command should always be readonly of course.

**Review by arpad-m — APPROVED** [review]

File: s3_scrubber/src/lib.rs [hunk 1]
```diff
@@ -86,7 +86,9 @@ impl S3Target {
         if new_self.prefix_in_bucket.is_empty() {
             new_self.prefix_in_bucket = format!("/{}/", new_segment);
         } else {
-            let _ = new_self.prefix_in_bucket.pop();
+            if new_self.prefix_in_bucket.ends_with('/') {
+                let _ = new_self.prefix_in_bucket.pop();
```
Comment (koivunej) [inline s3_scrubber/src/lib.rs:90]:
```suggestion
                new_self.prefix_in_bucket.pop().unwrap();
```

**Comment by koivunej** [comment]
Did the chaos start to fail because of the `/` change?

**Pull request merged**